File: java-samples/corda5-negotiation-cordapp/workflows/src/main/java/com/r3/developers/samples/negotiation/workflows/propose/ProposalFlowArgs.java
package com.r3.developers.samples.negotiation.workflows.propose;

import net.corda.v5.base.annotations.CordaSerializable;
import net.corda.v5.base.types.MemberX500Name;

@CordaSerializable
public class ProposalFlowArgs {
    private int amount;

    private MemberX500Name counterParty;

    private boolean isBuyer;

    public ProposalFlowArgs() {
    }

    public ProposalFlowArgs(int amount, MemberX500Name counterParty, boolean isBuyer) {
        this.amount = amount;
        this.counterParty = counterParty;
        this.isBuyer = isBuyer;
    }

    public int getAmount() {
        return amount;
    }

    public MemberX500Name getCounterParty() {
        return counterParty;
    }

    public boolean getIsBuyer() {
        return isBuyer;
    }
}

File: kotlin-samples/corda5-negotiation/workflows/src/main/kotlin/com/r3/developers/samples/negotiation/workflows/Propose/ProposalFlowArgs.kt
package com.r3.developers.samples.negotiation.workflows.Propose

import net.corda.v5.base.annotations.CordaSerializable
import net.corda.v5.base.types.MemberX500Name

@CordaSerializable
data class ProposalFlowArgs(
    var amount: Int,
    var counterParty: MemberX500Name,
    var isBuyer:Boolean

)
